File: 167.two-sum-ii-input-array-is-sorted.cpp
/*
 * @lc app=leetcode id=167 lang=cpp
 *
 * [167] Two Sum II - Input Array Is Sorted
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n=numbers.size(),i=0,j=n-1;
        while(i<j)
        {
            if(numbers[i]+numbers[j]==target)
            {
                return {i+1,j+1};
            }
            else if(numbers[i]+numbers[j]>target)
            {
                j--;
            }
            else {
                i++;
            }
        }
        return{-1};
    }
};
// @lc code=end

File: 167.two-sum-ii-input-array-is-sorted.py
#
# @lc app=leetcode id=167 lang=python3
#
# [167] Two Sum II - Input Array Is Sorted
#

# @lc code=start
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        start = 0
        end = len(numbers) - 1

        while start < end:
            sumVal = numbers[start] + numbers[end]
            if  sumVal== target:
                return [start + 1, end + 1]
            elif sumVal > target:
                end -= 1
            else:
                start += 1

        return [-1, -1]

# @lc code=end

